airbyte-integrations/connectors/source-gitlab/source_gitlab/source.py

```diff
@@ -43,7 +43,7 @@ def _generate_main_streams(self, config: Mapping[str, Any]) -> Tuple[GitlabStrea
         auth = TokenAuthenticator(token=config["private_token"])
         auth_params = dict(authenticator=auth, api_url=config["api_url"])
 
-        pids = list(filter(None, config.get("projects").split(" ")))
+        pids = list(filter(None, config.get("projects", "").split(" ")))
```

Comment:
This solve the problem with NoneType but if user doesn't specify the projects in the spec will break anyway in other part of the code. Maybe makes the `projects` required in spec.json OR have a function similar to  `_get_group_list` the user have access to.